server/main.cpp

```diff
@@ -0,0 +1,178 @@
+#include "uvw.hpp"
+#include <memory>
+#include <iostream>
+#include <random>
+#include "connection.hpp"
+#include "packets.hpp"
+
+constexpr const char *addr = "127.0.0.1";
+constexpr const int port = 4242;
+
+void acceptClient(uvw::TCPHandle &srv);
+
+template<Packet T>
+void broadcast(uvw::Loop &loop, std::string &game_id, const T &packet);
+
+void handleLogin(uvw::TCPHandle &handle, PacketReader &reader);
+
+std::string genGameId() {
+    int len = 4;
+    static auto &chrs = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+
+    thread_local static std::mt19937 rg{std::random_device{}()};
+    thread_local static std::uniform_int_distribution<std::string::size_type> pick(0, sizeof(chrs) - 2);
```

Comment:
chars powinny pyć const, bez tego może być na nich mutex. To nie ma powodu być threadlocal - po to tworzysz tutaj własny generator, żeby nie musieć go dzielić. I tak jest na stosie, i tak nie potrzeba reprodukowalności.

Reply:
syndrom stack overflow XD fixed